`LoadModel.Destroy` panics on nil `GoMLXModel`/`ORTModel` (double-close, partial init)

## Summary

`backends.LoadModel` builds a `model.Destroy` closure that calls `model.GoMLXModel.Destroy()` (or `model.ORTModel.Destroy()`) without nil-checking. If the model field is ever nil at the time `Destroy()` is invoked, the closure panics with a nil pointer dereference.

This is reachable in two real-world paths:

1. **Double-close.** The Destroy closure sets `model.GoMLXModel = nil` (or `ORTModel = nil`) at the end of a successful destroy. Calling `Destroy()` a second time then panics. Frameworks that call Close on every processor at flow stop and again at stream teardown (e.g. Bento) hit this every shutdown.
2. **Partial init.** If `LoadModel` fails after assigning the Destroy closure but before assigning `GoMLXModel` / `ORTModel`, callers that invoke Destroy in their cleanup path hit a nil deref.

## Reproduction

Embed Hugot in a long-running framework that calls `processor.Close()` more than once for the same loaded model (Bento's pipeline does this on stream teardown).

Stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x30]

github.com/knights-analytics/hugot/backends.LoadModel.func1()
    /backends/model.go:82 +0xe8
```

(Line numbers from v0.6.4; the same code path exists unchanged in v0.7.2.)

## Affected versions

Confirmed present in v0.6.4 and v0.7.2 (current). Same code shape across the v0.6.x and v0.7.x lines.

## Suggested fix

`backends/model.go` around line 72-86 — guard each backend's Destroy call:

```go
model.Destroy = func() error {
    var destroyErr error
    if model.Tokenizer != nil {
        destroyErr = model.Tokenizer.Destroy()
        model.Tokenizer = nil
    }
    switch options.Backend {
    case "ORT":
        if model.ORTModel != nil {
            destroyErr = errors.Join(destroyErr, model.ORTModel.Destroy())
            model.ORTModel = nil
        }
    case "GO", "XLA":
        if model.GoMLXModel != nil {
            model.GoMLXModel.Destroy()
            model.GoMLXModel = nil
        }
    }
    return destroyErr
}
```

Also nil-clears `Tokenizer` for the same double-close reason.

--- a/backends/model.go
+++ b/backends/model.go
@@ -74,14 +74,19 @@ func LoadModel(ctx context.Context, path string, onnxFilename string, options *o
 		var destroyErr error
 		if model.Tokenizer != nil {
 			destroyErr = model.Tokenizer.Destroy()
+			model.Tokenizer = nil
 		}
 		switch options.Backend {
 		case "ORT":
-			destroyErr = errors.Join(destroyErr, model.ORTModel.Destroy())
-			model.ORTModel = nil
+			if model.ORTModel != nil {
+				destroyErr = errors.Join(destroyErr, model.ORTModel.Destroy())
+				model.ORTModel = nil
+			}
 		case "GO", "XLA":
-			model.GoMLXModel.Destroy()
-			model.GoMLXModel = nil
+			if model.GoMLXModel != nil {
+				model.GoMLXModel.Destroy()
+				model.GoMLXModel = nil
+			}
 		}
 		return destroyErr
 	}

